src/components/Modal/Modal.tsx

```diff
@@ -176,12 +176,12 @@ export const ModalContent = (props: ModalContentProps) => {
  * </Modal>
  */
 export const Modal = (props: ModalProps) => {
-  const { ariaLabel, initialFocus, onClose, open, ...rest } = props;
+  const { 'aria-label': ariaLabel, initialFocus, onClose, open, ...rest } = props;
   const { children } = rest;
 
   if (process.env.NODE_ENV !== 'production') {
     const hasModalTitle = childrenHaveModalTitle(children);
-    if (!hasModalTitle && !ariaLabel) {
+    if (!hasModalTitle && !props['aria-label']) {
```

Comment:
Nit: because you're import it as `'aria-label': ariaLabel` on line 179, you shouldn't need to change this line or line 203 I think.